amp_iframe.html

```diff
@@ -4,67 +4,76 @@
   <meta charset="utf-8">
   <script src="//player.ooyala.com/static/v4/stable/latest/core.min.js"></script>
   <script src="//player.ooyala.com/static/v4/stable/latest/video-plugin/main_html5.min.js"></script>
+  <script src="//player.ooyala.com/static/v4/stable/latest/video-plugin/bit_wrapper.min.js"></script>
   <script src="//player.ooyala.com/static/v4/stable/latest/skin-plugin/html5-skin.min.js"></script>
   <link rel="stylesheet" href="//player.ooyala.com/static/v4/stable/latest/skin-plugin/html5-skin.min.css"/>
 </head>
 
 <body style="padding:0; margin:0; width:100%; height:100%; overflow:hidden">
   <div id='container' style='width:100%; height:100%;'></div>
   <script>
-    var queryParams = window.location.href;
-    var playerParam = {
-      "pcode": queryParams.match(/pcode=([^&]*)/)[1],
-      "playerBrandingId": queryParams.match(/pbid=([^&]*)/)[1],
-      "skin": {
-        "config": "",
-        "inline": {"shareScreen": {"embed": {"source": "<iframe width='640' height='480' frameborder='0' allowfullscreen src='//player.ooyala.com/static/v4/stable/latest/skin-plugin/iframe.html?ec=<ASSET_ID>&pbid=<PLAYER_ID>&pcode=<PUBLISHER_ID>'></iframe>"}}}
-      }
-    };
-
-    // Parse docUrl
-    var docUrl = undefined;
-    if (!!queryParams.match(/docUrl=([^&]*)/)) {
-      playerParam["docUrl"] = decodeURIComponent(queryParams.match(/docUrl=([^&]*)/)[1]);
-    }
+    function queryStringToJSON(qs) {
+      var pairs = qs.split('&');
+      var result = {};
+      pairs.forEach(function(pair) {
+        var pair = pair.split('=');
+        var key = pair[0];
+        var value = decodeURIComponent(pair[1] || '');
+        if (value === "true") {
```

Comment:
It doesn't look to me that switch here is feasible as these are some of conditions are logical, not string comparisons.